### PR TITLE
Update Solidity files in docs

### PR DESCRIPTION
**Pull request by ernestognw**
Found a couple of issues in the Solidity files from the documentation. I think it's easy to forget to update the files if they're not compiled.

I moved those files I thought convenient and fixed a couple of compilation issues. Also took the freedom to replace error strings with custom errors since that should be our go-to recommendation

#### PR Checklist

- [ ] Tests
- [x] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: contracts/mocks/docs/utilities/Base64NFT.sol [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    // ...
+
+    function tokenURI(uint256 tokenId) public pure override returns (string memory) {
+        // Equivalent to:
+        // {
+        //   "name": "Base64NFT #1",
+        //   // Replace with extra ERC-721 Metadata properties
+        // }
+        bytes memory dataURI = abi.encodePacked("{", '"name": "Base64NFT #', tokenId.toString(), '"', "}");
```
Comment (Amxx) [inline contracts/mocks/docs/utilities/Base64NFT.sol:23]: In these kind of application, I think using `string.concat` is clearer.
Also, things can be merges before.
```suggestion
        string memory dataURI = string.concat("{\"name\": \"Base64NFT #", tokenId.toString(), "\"}");
```

File: contracts/mocks/docs/utilities/Base64NFT.sol [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+        return string(abi.encodePacked("data:application/json;base64,", Base64.encode(dataURI)));
```
Comment (Amxx) [inline contracts/mocks/docs/utilities/Base64NFT.sol:25]:
```suggestion
        return string.concat("data:application/json;base64,", Base64.encode(dataURI));
```

**Comment by ernestognw** [comment]
I tried the suggestions but they didn't compile. I could make it compile but since it doesn't have tests, I need to make sure they produce the same output. Will do tomorrow.

**Comment by Amxx** [comment]
We ,eed to find a way to make tests and linting pass, otherwise this is going to affect all other PRs.

File: contracts/mocks/docs/utilities/Base64NFT.sol [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
```
Comment (Amxx) [inline contracts/mocks/docs/utilities/Base64NFT.sol:5]: We should not need the `@openzeppelin/contracts`. The `../../..` are supposed to be replaced automatically during doc generation

**Comment by ernestognw** [comment]
The linter fails because there's a conflict between a rule configured for Prettier to avoid single quotes in Solidity files:

```json
{
  ...
  "singleQuote": true,
  "overrides": [
    {
      "files": "*.sol",
      "options": {
        "singleQuote": false
      }
    }
  ],
  "plugins": ["prettier-plugin-solidity"]
}
```

So whenever we run `npm run lint:fix`, it makes the following:

```diff
-        string memory dataURI = string.concat("{\"name\": \"Base64NFT #", tokenId.toString(), "\"}");
+        string memory dataURI = string.concat('{"name": "Base64NFT #', tokenId.toString(), '"}');
```

Which then conflicts with the override for Solidity files.

I added a `prettier-ignore` comment since I found it the less intrusive way to make the CI pass without affecting any other configuration.

**Review by Amxx — APPROVED** [review]

**Pull request merged**